### PR TITLE
Use proper controller names in settings

### PR DESCRIPTION
**Pull request by eros71-dev**
Make the controller names show up in the selection box instead of things like 0, 1, etc... Now it will show names like "PS4 Controller", if provided by SDL2.

### CODE REVIEW

**Comment by PeachyPeachSM64** [comment]
not sure if it's a github thing but the indentation looks off

**Comment by Flower35** [comment]
> not sure if it's a github thing but the indentation looks off

It is off by one level (4 spaces) to the left, compared to the previous code block.

**Comment by eros71-dev** [comment]
Since git was messing me up a little with my stashed f3dex3 changes I ended up copying that block somewhere else, resetting the repo locally, and then pasted it back so there's a chance I forgot to clean up the indentation afterwards.

I'll change it in a bit.

File: src/pc/djui/djui_panel_controls.c [hunk 1]
```diff
@@ -25,21 +25,34 @@ void djui_panel_controls_create(struct DjuiBase* caller) {
...
+                char newName[256];
+                snprintf(newName, sizeof(newName), "%s (%d)", gamepadChoices[i], count);
+                free(gamepadChoices[i]);
+                gamepadChoices[i] = strdup(newName);
             }
-            sprintf(&gamepadChoicesLong[index], "%d", i);
-            gamepadChoices[i] = &gamepadChoicesLong[index];
         }
-        djui_selectionbox_create(body, DLANG(CONTROLS, GAMEPAD), gamepadChoices, numJoys, &configGamepadNumber, NULL);
+    }
```
Comment (PeachyPeachSM64) [inline src/pc/djui/djui_panel_controls.c]: Can crash due to a use-after-free bug + djui choices are never updated
This block should be moved before `djui_selectionbox_create`

**Review by PeachyPeachSM64 — CHANGES_REQUESTED** [review]

**Comment by eros71-dev** [comment]
Alright, everything should be okay now, from what I see, although I still need someone to make sure the name trimming for repeated controllers works properly.

File: src/pc/djui/djui_panel_controls.c [hunk 2]
```diff
@@ -28,18 +28,53 @@ void djui_panel_controls_create(struct DjuiBase* caller) {
...
+                    
+                    // If the name is bigger than 9 characters, we need to truncate it first
+                    // Then we can append the number so it fits in the slot
+
+                    // Should we look into making scrolling text for this?
+                    if (strlen(gamepadChoices[i]) > 9) {
+                        strncpy(newName, gamepadChoices[i], 9);
+                        newName[9] = '\0';
+                        strcat(newName, "...");
+                        snprintf(newName, sizeof(newName), "%s (%d)", newName, count);
```
Comment (PeachyPeachSM64) [inline src/pc/djui/djui_panel_controls.c]: using `newName` in `snprintf` as both the destination and an argument is undefined behavior in C
but you don't need to do that, printf functions have a specific format for trimming:
```
snprintf(newName, sizeof(newName), "%.9s... (%d)", gamepadChoices[i], count);
```

**Review by PeachyPeachSM64 — CHANGES_REQUESTED** [review]

**Comment by PeachyPeachSM64** [comment]
also I like the idea of scrolling text

File: src/pc/djui/djui_panel_controls.c (hunk 2 above)
Comment (eros71-dev) [reply src/pc/djui/djui_panel_controls.c]: Alright, changed it, hopefully I did it right heh

Comment (PeachyPeachSM64) [reply src/pc/djui/djui_panel_controls.c]: keep the if statement, otherwise you'll have the ellipsis even for names that have less than 9 characters

Comment (eros71-dev) [reply src/pc/djui/djui_panel_controls.c]: Ah, right my bad

**Comment by PeachyPeachSM64** [comment]
looks good to me

**Pull request merged**